### ISSUE TITLE
EventAllDay is not show after notifydatasetChanged()

### ISSUE THREAD

**Issue by kingvhit** [thellmund/Android-Week-View#91]
**Describe the bug**
Related issue #88. Please see the reproduce that I already try on your sample app.
It only occurs when async fetching data and call `notifydataSetChanged()`

**To Reproduce**
1. Go to `ApiEvent` class
2. Change `.setAllDay(true)` to all day fetching or set to some specific date if you want.
3. Go to `AsyncActivity`, waiting for loading data from `asset` and then `notifydataSetchanged` after done.
4. After progressing bar is gone, meaning that the data is available but, the event all day is not shown. If you want it show, you must scroll left or right and then it shows all of the events.

The problems may be related to `notifydatasetchanged`. 

**Expected behavior**
Other issues, I think to go to date work perfectly but it need run on delay thread. Do you know how I can do it immediately? Because I want to move it very fast if no the user will be seen the today show up first.

**Screenshots**
![1](https://user-images.githubusercontent.com/9419185/61735413-192ff980-adae-11e9-9d5c-edc05e2c8899.png)
![2](https://user-images.githubusercontent.com/9419185/61735415-192ff980-adae-11e9-8c56-dac68ce8ada6.png)
![3](https://user-images.githubusercontent.com/9419185/61735417-19c89000-adae-11e9-9269-e35967f69129.png)

**Comment by thellmund** [comment]
I was able to reproduce the issue. It has to do with `Updater.isRequired`: The way it was implemented in `AllDayEventsUpdater` and `HeaderRowHeightUpdater` worked when immediately showing data (e.g. `BaseActivity`), but not when showing data with a delay (e.g. `AsyncActivity`). 

I have committed a quick workaround (commit e57c14572a15a643ebe8bf3aac41bedc504bfe23), but will take a closer look at it again before releasing version 4.0.
